app/services/doc_auth/lexis_nexis/error_generator.rb

```diff
@@ -0,0 +1,105 @@
+        'Issue Date Valid': { type: ID, msg_key: 'doc_auth.errors.lexis_nexis.issue_date_checks' },
+        # i18n-tasks-use t('doc_auth.errors.lexis_nexis.id_not_verified')
+        'Layout Valid': { type: ID, msg_key: 'doc_auth.errors.lexis_nexis.id_not_verified' },
+        # i18n-tasks-use t('doc_auth.errors.lexis_nexis.id_not_verified')
+        'Near-Infrared Response': { type: ID, msg_key: 'doc_auth.errors.lexis_nexis.id_not_verified' },
+        # i18n-tasks-use t('doc_auth.errors.lexis_nexis.sex_check')
+        'Sex Crosscheck': { type: ID, msg_key: 'doc_auth.errors.lexis_nexis.sex_check' },
+        # i18n-tasks-use t('doc_auth.errors.lexis_nexis.visible_color_check')
+        'Visible Color Response': { type: ID, msg_key: 'doc_auth.errors.lexis_nexis.visible_color_check' },
+        # i18n-tasks-use t('doc_auth.errors.lexis_nexis.id_not_verified')
+        'Visible Pattern': { type: ID, msg_key: 'doc_auth.errors.lexis_nexis.id_not_verified' },
+        # i18n-tasks-use t('doc_auth.errors.lexis_nexis.visible_photo_check')
+        'Visible Photo Characteristics': { type: FRONT, msg_key: 'doc_auth.errors.lexis_nexis.visible_photo_check' },
+      }.freeze
+      # rubocop:enable Layout/LineLength
+
+      # rubocop:disable Metrics/PerceivedComplexity
+      def self.generate_trueid_errors(response_info, liveness_checking_enabled)
+        errors = Hash.new { |hash, key| hash[key] = Set.new }
+
+        if response_info[:DocAuthResult] != 'Passed'
+          response_info[:Alerts]&.each do |alert|
+            alert_msg_hash = TRUE_ID_MESSAGES[alert[:name].to_sym]
+
+            if alert_msg_hash.present?
+              # Don't show alert errors if the DocAuthResult has passed
+              # With liveness turned on DocAuthResult can pass but the liveness check fails
+              if alert[:result] != 'Passed'
+                # We should log the counts of failing alerts that are given to users for analytics
+                # AM: Log to Ahoy/Cloudwatch
+                errors[alert_msg_hash[:type]].add(I18n.t(alert_msg_hash[:msg_key]))
+              end
+              # else
+              # We always want to make sure any unknown alerts that come through are noted
+              # AM: Log to Ahoy/Cloudwatch
+            end
+          end
+        elsif liveness_checking_enabled && response_info[:PortraitMatchResults].present?
+          # Only bother to look for selfie_results if ID Auth was successful
+          if response_info[:PortraitMatchResults].dig(:FaceMatchResult) != 'Pass'
+            errors[:selfie].add(I18n.t('doc_auth.errors.lexis_nexis.selfie_failure'))
+            # else
+            # Should probably log if selfie results isn't populated when we expect it to be?
+          end
+        end
+
+        if errors.empty?
+          e = UnknownTrueIDError.new('TrueID failure escaped without useful errors')
+          # AM: Log to Ahoy/Cloudwatch
```

Comment:
I don't think these ahoy/cloudwatch comments make too much sense here. This object is passed to `error_messages` in the response class, and we have facilities to log `DocAuth::Response` objects and `FormResponse` objects, so we should just make sure the info we would want to log there is present here, right?

Reply:
We have the facilities to merge form responses, etc etc so if there is extra info we want to return here, we could just return it in a `DocAuth::Response` instance's extra, and then merge it on the way back up to make sure it gets logged